### PR TITLE
improvement: fern init

### PR DESCRIPTION
**Pull request by dsinghvi**
`fern init` is run at the root of users directory and generates a `fern.config.json`, `fern-api` directory, with a `sample api` and a sample `.fernrc.yml`

### CODE REVIEW

**Comment by dsinghvi** [comment]
@zachkirsch things we need to figure out are: 
1) Should we have a form that asks users what languages they want to generate in initially? 
2) What should we name folder & api in .fernrc.yml (considering that the sample apis are food delivery apis)?

File: packages/init/src/init.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+export async function initialize(): Promise<void> {
+    await writeProjectConfigIfNotExists();
+    await writeApiSubDirectoryIfNotExists();
+}
+
+const API_DIRECTORY = "fern-api";
+const SRC_DIRECTORY = "src";
+const API_SRC_DIRECTORY = `${API_DIRECTORY}/${SRC_DIRECTORY}`;
+const API_WORKSPACE_DEFINITION_FILE = `${API_DIRECTORY}/src/${WORKSPACE_DEFINITION_FILENAME}`;
```
Comment (zachkirsch) [inline packages/init/src/init.ts:19]: can also use path.join

File: packages/init/src/init.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        await writeSampleApiToDirectory(API_WORKSPACE_DEFINITION_FILE);
+    }
+}
+
+async function writeWorkspaceDefinitionFile(dir: string): Promise<void> {
+    const workspaceDefinition: WorkspaceDefinitionSchema = {
+        name: "Food Delivery API",
+        input: SRC_DIRECTORY,
+        plugins: [],
+    };
```
Comment (zachkirsch) [inline packages/init/src/init.ts:35]: you can pull this out if you want into a global like you did with DEFAULT_PROJECT_CONFIG

File: packages/init/src/sampleApi.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
+import { RawSchemas } from "@fern-api/syntax-analysis";
+import { writeFile } from "fs/promises";
+import yaml from "js-yaml";
+
+export async function writeSampleApiToDirectory(dir: string): Promise<void> {
+    await writeFile(`${dir}/${ORDER_API_FILENAME}`, yaml.dump(ORDER_API));
+    await writeFile(`${dir}/${MENU_API_FILENAME}`, yaml.dump(MENU_API));
```
Comment (zachkirsch) [inline packages/init/src/sampleApi.ts:7]: path.join

File: packages/init/src/init.ts (hunk 1 above)
Comment (zachkirsch) [reply packages/init/src/init.ts:19]: it's better cross platform, e.g. on windows when the separator is `\`

File: packages/init/src/init.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    await writeFile(dir, yaml.dump(workspaceDefinition));
+}
+
+const DEFAULT_PROJECT_CONFIG: ProjectConfigSchema = {
+    workspaces: ["**"],
+};
+
+async function writeProjectConfigIfNotExists(): Promise<void> {
+    if (!doesPathExist(PROJECT_CONFIG_FILENAME)) {
+        await writeFile(PROJECT_CONFIG_FILENAME, JSON.stringify(DEFAULT_PROJECT_CONFIG));
```
Comment (zachkirsch) [inline packages/init/src/init.ts:45]: `JSON.stringify(DEFAULT_PROJECT_CONFIG, undefined, 4)` will pretty-print it

File: packages/init/src/init.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    }
+}
+
+export async function doesPathExist(filepath: string): Promise<boolean> {
```
Comment (zachkirsch) [inline packages/init/src/init.ts:49]: does this need to be exported?

File: packages/init/src/sampleApi.ts [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+                        parameters: {
+                            restaurantId: "RestaurantId",
+                        },
+                        response: "list<MenuItem>",
+                    },
+                },
+            },
+        },
+    },
+};
```
Comment (zachkirsch) [inline packages/init/src/sampleApi.ts:120]: not super into to `init` dumping a big yaml, let's chat when you come in

File: packages/init/src/init.ts (hunk 1 above)
Comment (dsinghvi) [reply packages/init/src/init.ts:19]: thanks

File: packages/init/src/init.ts (hunk 5 above)
Comment (dsinghvi) [reply packages/init/src/init.ts:49]: nope

**Pull request merged**